GirafRest/Controllers/WeekController.cs

```diff
@@ -127,6 +127,25 @@ public async Task<ActionResult> ReadUsersWeekSchedule(string userId, int weekYea
                             var timerPlace = _giraf._context.Timers.FirstOrDefault(t => t.Key == activity.TimerKey);
                             activity.Timer = timerPlace;
                         }
+
+                        if (activity.Pictograms != null)
+                        {
+                            foreach (var pictogramRelation in activity.Pictograms)
+                            {
+                                var dbPictogram =
+                                    _giraf._context.Pictograms.FirstOrDefault(p => p.Id == pictogramRelation.PictogramId);
+                                if (dbPictogram != null)
+                                {
+                                    pictogramRelation.Pictogram =

```

Comment:
It looks like you make a redundant database call, insted of the secound call you could write 
```suggestion
                                    pictogramRelation.Pictogram = dbpictogram

```
Or is there a meaning with this?

Reply:
This comment turned out ugly, sorry. Hope it still makes sense